WPMedia/fusion-news-theme-blocks
# TMEDIA-442: Add Password Reset block

**If you have not filled out the checklist below, the pr is not ready for review.**

## Description
Added password Reset block

## Jira Ticket
- [TMEDIA-442](https://arcpublishing.atlassian.net/browse/TMEDIA-442)

## Acceptance Criteria
* Create a new block “Reset Password” that can be dragged onto a new page ‘/account/reset-password/'
* Reset Password block uses icon from the Icon Library of ‘redo’
* Block includes Headline (Create your new password), Subtext (Enter a new password for your account), password and confirm password fields (as used on Sign Up block) and Submit (Continue) button.  All words should be translatable strings so we can add other languages later.
* The passwords should match and follow the minimum password requirements, otherwise an error state is shown (TMEDIA-439). This is already working on the Sign Up page so we should re-use exactly that.
* On click of Submit button, success message shows Headline (New password saved) and Subtext (Your new password has been saved) and Continue button.  These should be translatable strings so we can add other languages later.
  * Success button text and URL should be Custom Fields of the block with default of “Continue to Homepage” and “/” as the defaults

## Test Steps
1. Checkout this branch `git checkout tmedia-442`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/identity-block`
3. Add a new page ( Account / Reset Password :: /account/reset-password/ ):
![image](https://user-images.githubusercontent.com/2287238/148305292-91946c9a-f774-4463-a371-307ef43933df.png)
4. Use a Single Column Layout
5. Add the Navigation fragment to the the first section:
![image](https://user-images.githubusercontent.com/2287238/148305475-ffa7b85f-4042-49f7-bffd-7d2470871c50.png)
6. Add the reset block to the second section:
![image](https://user-images.githubusercontent.com/2287238/148305528-ff050f50-a47f-46bf-b591-b14aafa8492d.png)
7. Verify Custom Field:
![image](https://user-images.githubusercontent.com/2287238/148305566-58a8af2d-7d84-4cc6-99f3-9ff212a664b0.png)
8.  Get `<your key>` from `arc-themes-feature-pack/.env` : `ARC_ACCESS_TOKEN`
9.  Run this websocket listener (`node ws.js`):
```js
    var WebSocket = require('ws')

    var options = {
      headers: {
        "Arc-Site": "arc-demo-5",
        "Authorization": "Bearer <your key>"
      }
    };

    const connection = new WebSocket('wss://api.corecomponents.arcpublishing.com/retail/api/v1/event/stream/0', options)

    connection.onopen = () => {
      setInterval(function () {
        connection.send('ping');
      }, 5000);
      console.log('Connection Established');
    }

    connection.onerror = error => {
      console.log(`WebSocket error: ${error.message}`);
    }

    connection.onmessage = e => {
      var obj = JSON.parse(e.data);
      if(obj.type !== 'PONG'){
        console.log('-'.repeat(90));
        console.log(obj);

        const { type, message } = obj;
        switch (type) {
          case 'PASSWORD_RESET_REQUEST': {
            const { nonce } = message;
            console.log(`http://localhost/pf/account/reset-password/?_website=arc-demo-5&nonce=${nonce}`);
          }
        }
        console.log('-'.repeat(90));
      }
    }
```
10. Visit: http://localhost/pf/account/forgot-password/?_website=arc-demo-5 (forgot password page from prior PR!)
11. Enter your valid email and request a reset and you should see something like this in your terminal:
![image](https://user-images.githubusercontent.com/2287238/148306022-5fdcbada-59aa-41e5-9b0c-caefcc66ba2b.png)
12. Copy the URL provided and navigate to that.
13. Enter a new password, confirm, and submit that.
14. You should see another request come into the websocket listener:
![image](https://user-images.githubusercontent.com/2287238/148306179-7cf12a85-fbb5-484e-991f-32b1d6f65792.png)

## Effect Of Changes
### After
![image](https://user-images.githubusercontent.com/2287238/148306235-59b05e97-56be-407f-a80d-4b8b10d2460d.png)


## Dependencies or Side Effects
N/A

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
